db/migration/1745521253422-DodsInAdmin.ts

```diff
@@ -0,0 +1,223 @@
+                console.error(
+                    `Error: Cannot migrate dods. Gdoc with id ${GDOCS_DETAILS_ON_DEMAND_ID} not found`
+                )
+                return
+            }
+            const content = JSON.parse(dodGdoc.content)
+            const rawDetails = content.details
+            const parsedDetails = DEPRECATED_parseDetails(rawDetails)
+            return parsedDetails
+        })
+
+    if (!details?.details) {
+        console.error(
+            `Error: Cannot migrate dods. Details are not set or empty`
+        )
+        return
+    }
+
+    console.log(
+        `Inserting ${Object.values(details.details).length} dods into db...`
+    )
+
+    for (const [name, detail] of Object.entries(details.details)) {
+        const asMarkdown = enrichedBlocksToMarkdown(detail.text, false)
+        if (!asMarkdown) {
+            console.error(`Failed to convert ${name} to markdown`)
+            continue
+        }
+
+        await queryRunner.query(
+            `-- sql
+            INSERT INTO dods (name, content, lastUpdatedUserId)
+            VALUES (?, ?, ?)
+        `,
+            [name, asMarkdown, 1]
+        )
+
+        const id = await queryRunner
+            .query(
+                `-- sql
+            SELECT id FROM dods WHERE name = ?`,
+                [name]
+            )
+            .then((rows) => rows[0].id)
+
+        const plaintextLinks = extractLinksFromMarkdown(asMarkdown)
+        for (const [text, url] of plaintextLinks) {
+            const link = createLinkFromUrl({
+                url,
+                sourceId: id,
+                text,
+                componentType: "markdown",
+            })
+
+            await queryRunner.query(
+                `-- sql
+                INSERT INTO dod_links (sourceId, target, componentType, linkType, text, queryString, hash)
+                VALUES (?, ?, ?, ?, ?, ?, ?)
+            `,
+                [
+                    link.sourceId,
+                    link.target,
+                    link.componentType,
+                    link.linkType,
+                    link.text,
+                    link.queryString,
+                    link.hash,
+                ]
+            )
+        }
+    }
+    console.log(
+        `Inserting ${Object.values(details.details).length} dods into db...done`
+    )
+}
+
+export class DodsInAdmin1745521253422 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`-- sql
+            CREATE TABLE dods (
+                id INTEGER PRIMARY KEY AUTO_INCREMENT,
+                name VARCHAR(512) NOT NULL UNIQUE,
+                content VARCHAR(4096) NOT NULL,
+                createdAt TIMESTAMP NOT NULL DEFAULT now(),
+                updatedAt TIMESTAMP NOT NULL DEFAULT now(),
+                lastUpdatedUserId INTEGER NOT NULL,
+                FOREIGN KEY (lastUpdatedUserId) REFERENCES users(id) ON DELETE CASCADE
```

Comment:
Is cascading deletion what we want here? Does it not matter because we never delete users?